app/handlers/newflow/verify_email_by_pin.rb

```diff
@@ -0,0 +1,35 @@
+module Newflow
+    # Marks an `EmailAddress` as `verified` if it matches the passed-in `EmailAddress`'s pin
+    # and then marks the owner of the email address as 'activated'.
+    class VerifyEmailByPin
+    lev_handler
+    uses_routine ConfirmByPin
+
+    paramify :confirm do
+      attribute :pin, type: String
+
+      validates :pin, presence: true
+    end
+
+    def authorized?
+      true
+    end
+
+    # TODO: rate-limit this action here (or in the controller?)
```

Comment:
We determined during code review that ConfirmByPin already does rate-limiting via sequential_failures.

Reply:
👍 🎉